### DISCUSSION TITLE
Parent/Child Delete button not working

### DISCUSSION THREAD

**Discussion by george-luciditylab** [filamentphp/filament#9399] [Help]
Hi, I'm hoping someone may have an idea how to help with this.

I have set up a resource called 'Pages'. This resource gives users the ability to create and manage pages on their website which includes parent/child relationships.

I have a route setup as follows:
`'children' => ListNodePages::route('/{parent_id}/children'),`

When accessed, this route lists all pages with the parent id in the url. 

This is all working as expected, however, when I click the delete action on one of the child records the page refreshes and the confirm deletion modal does not appear.

If you need any more information, please let me know :) 

Thanks in advance!

**Comment by VincvanAs** [comment]
Maybe you could try to define the record thats needs to be deleted e.g. ->record($this->child) on the delete action

**Reply by george-luciditylab** [reply]
I have attempted to get this to work but with no joy. Here is a little more information to hopefully get an answer for this:

When I initially load the resource using the path '/admin/pages', all of my top-level pages are appearing correctly in the table. 

![top-level-page-listing](https://github.com/filamentphp/filament/assets/147987381/69ac8d3c-6844-40fa-9729-516ce2c3f364)

The delete action on each record is working as expected by triggering the confirmation modal.

![trigger-delete-action](https://github.com/filamentphp/filament/assets/147987381/d7a4dda6-5325-4818-ba25-e61319dccfb4)

If any of the top-level records have child pages, the record shows a red link with chevron down icon and links to '/admin/pages/{record_id}/children'. When I access this path, the child pages are listed correctly.

![about-us-child-pages](https://github.com/filamentphp/filament/assets/147987381/429cbc55-adb9-470d-9fb9-504128f18dbd)

Now, on this screen if I click a delete action on one of the two records, the delete confirmation modal does not appear and the table refreshes displaying the top-level records but the path in the browser remains the same (In this case '/admin/pages/2/children')

My eloquent query that returns the results for each page is as follows:
`
// define request variables
$record = request('record') ?? 0;
$request = request('parent_id') ?? 0;
return parent::getEloquentQuery()
            ->where('id', $record)
            ->orWhere('parent_id', $request);
`

My getPages function is:
'
return [
            'index' => ListNodePages::route('/'),
            'children' => ListNodePages::route('/{parent_id}/children'),
            'create' => CreateNodePages::route('/{parent_id}/create'),
            'edit' => EditNodePages::route('/{record}/edit'),
];
'

This same issue is also happening when trying to sort child pages. Clicking the sort action button when I am viewing the child pages '/admin/pages/2/children' results in the table changing to display all top-level pages rather than the child pages

I hope this all makes sense?
